[18.0][IMP] sign_oca: add kanban/activity view on sign.oca.request.signer

jguenat opened the pull request:
When using the sign systray, the user goes to a kanban view.

- There is no kanban view for this model so Odoo uses a simple incomprehensible one.
- No chatter models inherited on  sign.oca.request.signer so the activity view throws an error.

![sign_request_signer](https://github.com/user-attachments/assets/20479861-bc71-41af-96d7-617a0b554e76)

I propose to simply remove the kanban/activity views from the action. Tree and form view is enough IMO.

pedrobaeza commented:
Isn't better to provide both views instead?

jguenat commented:
Well it could be done but I took a look at previous versions and it wasn't there.
The action views kanaban & activity have been added in the JS during the V17 migration maybe by mistake?

pedrobaeza commented:
Yeah, maybe, but I think it's worth to add both views.

jguenat commented on sign_oca/views/sign_oca_request_signer.xml:
```diff
@@ -0,0 +1,131 @@
+                        <field name="signed_on" readonly="1" />
+                        <field name="inalterable_hash" />
+                        <field name="latitude" readonly="1" />
+                        <field name="longitude" readonly="1" />
```
I added readonly on these fields, makes to sens to edit them in form view right?

jguenat commented on sign_oca/views/sign_oca_request_signer.xml:
```diff
@@ -0,0 +1,131 @@
+                        type="object"
+                        name="sign"
+                        invisible="not is_allow_signature"
+                    />
```
Added class btn btn-primary for better UX

jguenat commented on sign_oca/views/sign_oca_request_signer.xml:
```diff
@@ -0,0 +1,131 @@
+                        <field name="longitude" readonly="1" />
+                    </group>
+                </sheet>
+                <chatter />
```
Not sure about the necessity of a chatter on request.signer, there is already on on the request maybe this will be confusing?

jguenat replied:
I added it because it what I usually do when addintg activity (& mail.thread) on the model

jguenat reviewed (commented):
Simple kanban,search and activity view added

some modifications on form view, see below comments

jguenat commented:
I created a new file for the sign.oca.request.signer views but that makes the PR confusing to read, let me know if you don't want that or I could do it in two separate commits

pedrobaeza reviewed (commented):
I think they can be in the same file as previously, as it's a children model.

pedrobaeza commented on sign_oca/views/sign_oca_request.xml:
```diff
@@ -339,20 +340,97 @@
                         <field name="partner_id" readonly="1" />
                         <field name="request_id" readonly="1" />
-                        <field name="signed_on" />
+                        <field name="signed_on" readonly="1" />
```
Why changing this?

jguenat replied:
it's my understanding that people should sign with the /sign_oca/document route which will take care of the date/latitude/longitude fields.

pedrobaeza replied:
Yeah, but this is unrelated to the PR, isn't it?

jguenat replied:
yeah this originally FIX pull request made to fix small issue when viewing request.signer kind of evolved to IMP pull request adding kanban/activity view so why not also fix the form view?

I can remove it no issue for me

pedrobaeza replied:
Well, at least it should be in a separate commit. What do you think, @etobella ?

etobella replied:
I will do it in another PR. on the migration, a lot of readonly where removed without sense by @peluko00 

https://github.com/OCA/sign/commit/bd3c52d992512785fb54e7dee30407a31af76e4d#diff-4a09fca0e1adda50e9bd55b26df637e63fe1ad2bb513aa2f6f2877df7eb9cbcfL354

It is something out of the scope of this PR.

pedrobaeza replied:
It's not without sense, but moved the readonly to views, as if not, you can't fill that fields through Odoo Excel import mechanism.

jguenat replied:
I removed the readonly change, kept the button primary on form view, renamed commit (could rename PR also)

jguenat committed:
[IMP] sign_oca: improve request.signer views
    - add kanban & search view
    - enable activity : add chatter on form view
    - sign button primary

victoralmau reviewed (approved):
Functional review OK

pedrobaeza commented:
/ocabot merge minor
